02_bubble/bubble.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, bubbleSort([]int{3, 2, 1, 5}))
+}
+
+func bubbleSort(array []int) []int {
+	arrayLength := len(array)
+	for i := 0; i < arrayLength; i++ {
+		for j := 0; j < arrayLength-i-1; j++ {
+			if array[j] > array[j+1] {
+				array[j], array[j+1] = array[j+1], array[j]
+			}
+		}
+	}
+	return array
```

Comment:
Is mutating the input slice permitted in the problem statement?

Reply:
Done